meteor/server/databaseMigrationSystemSteps.ts

```diff
@@ -102,6 +102,127 @@ function ensureCollectionProperty<T = any> (
 		}
 	}
 }
+/**
+ * Convenience function to generate basic test
+ * øparam type
+ */
+function ensurePlayoutDevicesHostAndPort (
+	deviceType: PlayoutDeviceType = -1
+): MigrationStepBase | null {
+	let collection: Mongo.Collection<any> = Collections.PeripheralDevices
+	if (!collection) throw new Meteor.Error(404, `Collection PeripheralDevices not found`)
+
+	let allDevices = {}
+
+	// iterate over playout gateways
+	collection.find({'type': PeripheralDeviceAPI.DeviceType.PLAYOUT}).fetch().forEach((playoutGw) => {
+		let devices: PlayoutDeviceSettings = objectPathGet(playoutGw, 'settings.devices', {})
+		// filter on device-type unless devicetype === -1
+		if (deviceType > -1) {
+			devices = _.pick(devices, (device) => device.type === deviceType)
+		}
+		allDevices[playoutGw._id] = devices
+	})
+
+	// iterate over devices for gateways
+	_.each(allDevices, (devices, key) => {
+		allDevices[key] = _.pick(devices, (device) => {
+			let needsMoreData = false
+			let foundHostProp = objectPathGet(device, 'options.host', undefined) 
```

Comment:
@nytamin same at :132 The MAJOR question is to require host and port fields to be present at all in the database before failing the validation. As of now only non-empty strings passes hosts, and only numbers > 0 passes ports.

If we DON'T require an initial entry to even consider validating the device for host or port, we would need a map of device types to search for. This will align better now that I'm introducing the DefaultPlayoutDevicePort constant, which I didn't have as I originally wrote this validation check.